File: simulation/f_x.m
function [F_X] = f_x(coor,J_2,A_e,GM)
    r = sqrt(coor(1,1)^2 + coor(2,1)^2 + coor(3,1)^2);
    coor_p = [coor(1,1)/r; coor(2,1)/r; coor(3,1)/r];
    GM_P = GM/(r^2);
    ro = A_e/r;
    F_x = [coor(4,1); coor(5,1); coor(6,1); ...
            -GM_P*coor_p(1,1) - (3/2)*J_2*GM_P*coor_p(1,1)*(ro^2)*(1-5*(coor_p(3,1)^2));...
            -GM_P*coor_p(2,1) - (3/2)*J_2*GM_P*coor_p(2,1)*(ro^2)*(1-5*(coor_p(3,1)^2));...
            -GM_P*coor_p(3,1) - (3/2)*J_2*GM_P*coor_p(3,1)*(ro^2)*(3-5*(coor_p(3,1)^2))];
        [F_X] = F_x;
end

File: libglnsvpos/Main/Main/f_x.cpp

#include <math.h>
#include <iostream>
#include <stdlib.h>
#include"f_x.h"

/*
Данная функция позводяет вычислить коэф. К 
для реализации метода Рунге-Кутты

Входные данные:
double A_e - большая полуось общеземного эллипсоида[м],
double GM - геоцентр.конст.грав.поля Земли с учетом атмосф.[м ^ 3 / c ^ 2]б
struct Place cor - структура PLace с поправками от коэф. K,
double J_2 - зональный гармонический коэф.второй степени.

Выходные данные:
struct Place К - структура коэф K.

*/

typedef struct Place
{
    double x;
    double y;
    double z;
    double V_x;
    double V_y;
    double V_z;

} Place;


typedef struct Cor
{
    double x;
    double y;
    double z;
} Cor;


struct Place f_x(double A_e, double GM,
         struct Place cor, double J_2)
{
        Cor cor_p;
        Place K;
        double r = sqrt(cor.x * cor.x + cor.y * cor.y + cor.z * cor.z);
        cor_p.x = cor.x / r;
        cor_p.y = cor.y / r;
        cor_p.z = cor.z / r;
        double GM_P = GM / (r * r);
        double ro = A_e / r;

        K.x = cor.V_x;
        K.y = cor.V_y;
        K.z = cor.V_z;
        K.V_x = - GM_P * cor_p.x - (3 / 2.0) * J_2 * GM_P * cor_p.x * (ro * ro) * (1 - 5 * (cor_p.z * cor_p.z));
        K.V_y = - GM_P * cor_p.y - (3 / 2.0) * J_2 * GM_P * cor_p.y * (ro * ro) * (1 - 5 * (cor_p.z * cor_p.z));
        K.V_z = - GM_P * cor_p.z - (3 / 2.0) * J_2 * GM_P * cor_p.z * (ro * ro) * (3 - 5 * (cor_p.z * cor_p.z));

        return K;
}